packages/compass-saved-aggregations-queries/src/components/aggregations-queries-list.tsx

```diff
@@ -97,21 +101,33 @@ const AggregationsQueriesList = ({
         case 'open':
           return onOpenItem(id);
         case 'delete':
-          return setDeletingItem(sortedItems.find((x) => x.id === id));
+          return onDeleteItem(id);
+        case 'copy':
+          return onCopyToClipboard(id);
       }
     },
-    [sortedItems, onOpenItem]
+    [onOpenItem, onDeleteItem, onCopyToClipboard]
   );
 
   const renderItem: React.ComponentProps<typeof VirtualGrid>['renderItem'] =
     useCallback(
-      ({ index }: { index: number }) => {
-        const item: Omit<SavedItemCardProps, 'onAction'> = sortedItems[index];
+      ({
+        index,
+        ...props
+      }: Omit<React.HTMLProps<HTMLDivElement>, 'type'> & { index: number }) => {
+        const item = sortedItems[index];
+
         return (
           <SavedItemCard
-            {...item}
+            id={item.id}
+            type={item.type}
+            name={item.name}
+            database={item.database}
+            collection={item.collection}
+            lastModified={item.lastModified}
```

Comment:
We were spreading things that don't belong here before (like the whole aggregation or query would be a passed prop), so I changed to explicitly passing props that are required